tasks/config.ts

```diff
@@ -155,24 +156,46 @@ export function load() {
     });
   });
 
-  async function getPriceFeedsConfig(hre: HardhatRuntimeEnvironment, deployer: SignerWithAddress) {
-    const priceFeeds = Object.entries(PriceFeedContractNames).map(async ([key, value]) => {
-      const instance = await hre.ethers.getContract(value, deployer);
-      return { symbol: key, address: instance.address, decimals: await getDecimals(instance) };
+  async function getPriceFeedsConfig(hre: HardhatRuntimeEnvironment, deployer: SignerWithAddress): Promise<{ [key: string]: PriceFeed }> {
+    // TODO - Multichain: This approach won't work when we're not the ones deploying the contract (for example when Chainlink is the one doing it) - https://lagoslabs.atlassian.net/browse/LAGO-890
+    const priceFeedsPromises = Object.entries(PriceFeedContractNames).map(async ([key, value]) => {
+      const priceFeedContract = await hre.ethers.getContract(value, deployer);
+      const priceFeedData: PriceFeed = {
+        address: priceFeedContract.address,
+        decimals: await getPriceFeedDecimals(priceFeedContract),
+        updatable: isPriceFeedUpdatable(priceFeedContract)
+      };
+
+      return { [key]: priceFeedData };
     });
 
-    const results = await Promise.all(priceFeeds);
-    return results;
+    const results = await Promise.all(priceFeedsPromises);
+
+    return results.reduce((previous, current) => {
+      const key = Object.keys(current)[0];
+      previous[key] = current[key];
+
+      return previous;
+    }, {});
   }
 
-  async function getDecimals(instance: Contract): Promise<number | undefined> {
+  function isPriceFeedUpdatable(priceFeed: Contract): boolean {
+    const isFiatPriceFeed = priceFeed.setPrice; // Taken from FiatToFiatPseudoOracleFeed
+    const isSovrynCryptoPriceFeed = priceFeed.updateOracleWithVerification; // Taken from SovrynSwapPriceFeed
+
+    return !!(isFiatPriceFeed || isSovrynCryptoPriceFeed);
+  }
+
+  // We have different ways on getting the decimals based on the price feed type
+  async function getPriceFeedDecimals(instance: Contract): Promise<number> {
     if (instance.decimals) {
       return await instance.decimals();
     } else if (instance.getDecimals) {
       return await instance.getDecimals();
-    } else {
-      return undefined;
     }
+
+    // TODO - Multichain: When adding support for a chain we're not the one who deployed the contract, then handle this case - https://lagoslabs.atlassian.net/browse/LAGO-890
+    return 0;
```

Comment:
shouldn't we throw an error instead?

Reply:
Can't really get here to be honest, and throwing an error will just mean I'll have to add undefined as a return value which will make the code ugly.. Let's leave it for now as it will be resolved in the ticket I mentioned above it